### PR TITLE
Initial commit for GaussianEstimateNoiser

### PR DESCRIPTION
**Pull request by pasin30055**
Implement GaussianMechanism and GeometricEstimateNoiser, which add Gaussian noise to a function/estimator to ensure differential privacy.

Note: currently the noise is calculated using a bound from Dwork-Roth book which is only valid for epsilon no more than one. A TODO is added to use a more accurate computation of sigma, which is also valid for all ranges of epsilon and delta.

### CODE REVIEW

File: src/common/noisers.py [hunk 1]
```diff
@@ -116,6 +116,52 @@ def __call__(self, x):
...
+      random_state:  Optional instance of numpy.random.RandomState that is
+        used to seed the random number generator.
+    """
+    self._func = f
+    self._delta_f = delta_f
+    # TODO(pasin30055): Use tight computation of sigma.
+    # Set the standard deviation sigma, following Appendix A of Dwork and Roth.
+    self._sigma = (math.sqrt(2 * math.log(1.25 / delta)) * delta_f *
+      math.sqrt(num_queries) / epsilon)
+    if random_state:
```
Comment (kungfucraig) [inline src/common/noisers.py:155]: self.random_state = random_state if random_state is not None else RandomState()

**Review by kungfucraig — COMMENTED** [review]
I think you mean GaussianEstimateNoiser, right?

**Comment by pasin30055** [comment]
> I think you mean GaussianEstimateNoiser, right?

File: src/common/noisers.py (hunk 1 above)
Comment (pasin30055) [reply src/common/noisers.py]: Thanks for the suggestion. I changed this to a more compact:
  self._random_state = random_state or np.random.RandomState()
and also made similar changes in LaplaceMechanism, GeometricMechanism.

**Comment by pasin30055** [comment]
> I think you mean GaussianEstimateNoiser, right?

I'm not sure whether you're referring to the title of the pull request, but I've changed it to GaussianEstimateNoiser. If you meant something else, please let me know.

**Review by jiayu-google — APPROVED** [review]
LGTM, thanks Pasin!

**Comment by pasin30055** [comment]
@jiayu-google Would you mind merging this request? (I don't have the write access...)

**Review by my2457 — APPROVED** [review]

**Comment by jiayu-google** [comment]
@pasin30055  Just merged.

**Pull request merged**